src/core/Wyam.Core/Modules/Control/FrontMatter.cs

```diff
@@ -109,30 +107,17 @@ public override IEnumerable<IDocument> Execute(IReadOnlyList<IDocument> inputs,
             List<IDocument> results = new List<IDocument>();
             context.ForEach(inputs, input =>
             {
-                List<string> inputLines = input.Content.Split(new[] { '\n' }, StringSplitOptions.None).ToList();
-                int delimiterLine = inputLines.FindIndex(x =>
-                {
-                    string trimmed = x.TrimEnd();
-                    return trimmed.Length > 0 && (_repeated ? trimmed.All(y => y == _delimiter[0]) : trimmed == _delimiter);
-                });
-                int startLine = 0;
-                if (delimiterLine == 0 && _ignoreDelimiterOnFirstLine)
-                {
-                    startLine = 1;
-                    delimiterLine = inputLines.FindIndex(1, x =>
-                    {
-                        string trimmed = x.TrimEnd();
-                        return trimmed.Length > 0 && (_repeated ? trimmed.All(y => y == _delimiter[0]) : trimmed == _delimiter);
-                    });
-                }
-                if (delimiterLine != -1)
+                Splitter splitter =
+                    new Splitter(_delimiter, _repeated)
+                        .IgnoreDelimiterOnFirstLine(_ignoreDelimiterOnFirstLine);
+
+                List<string> sections = splitter.Split(input.Content);
+
+                if (sections.Count == 2)
```

Comment:
See the comment above about the `Split()` return value - testing for a specific number of items here doesn't quite feel right.